spec/plans/util/retrieve_and_upload_spec.rb

```diff
@@ -0,0 +1,23 @@
+# spec/spec_helper.rb
+describe 'peadm::util::retrieve_and_upload' do
+  # Include the BoltSpec library functions
+  include BoltSpec::Plans
+
+  it 'file needs downloaded and needs uploaded' do
+    expect_command("test -e '/tmp/download'").error_with('kind' => 'nope', 'msg' => 'The command failed with exit code 1')
+    expect_task('peadm::download')
+    expect_task('peadm::filesize').be_called_times(2).return_for_targets(
+      'local://localhost' => { 'size' => '2'},
+      'primary' => { 'size' => 'null'}
+    )
+      expect_upload('/tmp/download').with_destination('/tmp/upload').with_params({}).return do |targets, source, destination, params|
+      results = targets.map do |target|
```

Comment:
Should there be some indentation here within the `do` blocks?

Reply:
Yes